shell/platform/linux/fl_task_runner.h

```diff
@@ -27,14 +28,14 @@ FlTaskRunner* fl_task_runner_new(FlEngine* engine);
 /**
  * fl_task_runner_post_task:
  * @task_runner: an #FlTaskRunner.
- * @task: Flutter task being scheduled
+ * @delegate: The callback to be scheduled
  * @target_time_nanos: absolute time in nanoseconds
  *
  * Posts a Flutter task to be executed on main thread. This function is thread
  * safe and may be called from any thread.
  */
 void fl_task_runner_post_task(FlTaskRunner* task_runner,
-                              FlutterTask task,
+                              std::function<void()> delegate,
```

Comment:
While the embedder code is complied as C++, the actual code is C. So I don't think using `std::function` here is a way to go. You can perhaps replace it with a callback that takes `gpointer` and a `gpointer` value.